File: api/src/main/java/org/roylance/yaas/services/IAuthenticationService.java
package org.roylance.yaas.services;

import org.roylance.yaas.models.YaasModels;

import java.util.List;

public interface IAuthenticationService {
    boolean userNameExists(final String userName);
    YaasModels.UIAuthentication authenticateUser(final YaasModels.UserModel user);
    YaasModels.UIAuthentication authenticateUser(final String token);
    YaasModels.UIAuthentication registerUser(final YaasModels.UserModel user);
    YaasModels.UIAuthentication changePassword(final YaasModels.UIChangePassword changePassword);
    boolean saveInfo(final YaasModels.UserModel user, final String token);

    // admin functions
    boolean changePasswordForUser(final YaasModels.UserModel user);
    List<YaasModels.UIAuthentication> getAllUsers(final int limit, final int offset);
    boolean isUserAdmin(final YaasModels.UserModel user);
    boolean setUserAsAdmin(final YaasModels.UserModel user);
    boolean removeUserAsAdmin(final YaasModels.UserModel user);
    boolean deleteUser(final YaasModels.UserModel user);
}

File: api/src/main/java/org/roylance/yaas/services/IAuthenticationService.kt
package org.roylance.yaas.services

import org.roylance.yaas.models.YaasModels

interface IAuthenticationService {
    fun exists(userName: String): Boolean
    fun login(user: YaasModels.User): YaasModels.UIAuthentication
    fun login(request: YaasModels.UIRequest): YaasModels.UIAuthentication
    fun register(user: YaasModels.User): YaasModels.UIAuthentication
    fun changePassword(model: YaasModels.UIChangePassword): YaasModels.UIAuthentication
    fun save(request: YaasModels.UIRequest): YaasModels.UIResponse
}